### ISSUE TITLE
Support Data layers (was: Cannot create new Polyline class)

### ISSUE THREAD

**Issue by vijay-joshi** [xkjyeah/vue-google-maps#90]
So, I want the user to be able to draw a freehand polygon on click of a button. I have it implemented in plain JS and now want to use it in vue (using vue-cli).
Here is the code I am using:

` 

      var self = this;
      // calls another method which disables map controls
      self.disableMap();
      // below line is fine
      console.log(self.$refs.map.$mapObject);
      // event handler is being called
      self.$refs.map.$mapObject.addListener('mousedown', function () {
        // the below line results in undefined
        console.log(self.$refs.$polylineObject);
        let poly = new (self.$refs.map.$polylineObject)({map: self.$refs.map, clickable: false});
      })
`
Any help greatly appreciated.

**Comment by xkjyeah** [comment]
Native Google Maps way:
```
Vue.component('polyline-generator', {
    mixins: [VueGoogleMaps.MapElementMixin],
    deferredReady() {
        this.$polyline = new google.maps.Polyline
        this.$mapObject.addListener('mousedown', (event) => {
            ...(omitted)...
            this.$emit('lineUpdated', this.$polyline.getPath())
        })
    },
}
```

If you want to do it in your current style it should look something like:
```html
<gmap-polyline ref="polyline">
```
```js
this.$refs.polyline.$polylineObject
```

The vue-ish style:
You will want to do something along the lines of:
```html
<gmap-map @mousedown="addPointToPath">
<gmap-polygon :path="path"></gmap-polygon>
</gmap-map>
```

```js
methods: { addPointToPath(event) {
    this.path.push({lat: event.latLng.lat(), lng: event.latLng.lng()})
}
```

Granted, this will be less performant than the native method, because you need to keep translating the Vue path into a Google Maps MVCArray

**Comment by davidvleung** [comment]
I'm just learning Vue and Google Maps but why the need to write your own code to free hand draw polygons?  Why not utilize the existing DrawingManager or google.map.data Layer?  Does the current architecture not lend to using them?

I'm mostly asking because I'm trying to do the same thing.  I've managed to turn on the data.controls (although not in a nice way) but struggling to figure how to incorporate the newly created drawings to my Vue instance.

**Comment by xkjyeah** [comment]
You make a very good point, and the embarrassing answer is that I didn't
know about google.maps.data when I started.

**Comment by xkjyeah** [comment]
So if you are happy to show me how you use the data layer, I could think
about you to wrap a Vue component around it

**Comment by devpake** [comment]
I actually deleted my comment because I finally started understanding how things are organized in vue-google-maps and managed to find my way to the data layer.

But here's what I ended up doing:  https://jsfiddle.net/devpake/onz7m5m1/1/
I basically took one of your examples and modified it slightly.

Main thing to note - the way vue-google-maps creates Polygons is via google.maps.Polygon and is thus, separate from the Data Layer.  So if someone were to create a Polygon like normal, it won't show up in the Data Layer's data set.  (i.e. toGeoJson would return empty).  You can see this by hitting the "Display Geo JSON" button before drawing anything.  You'll note that it doesn't display any data.  However, after loading the controls and drawing a polygon, there will then be data when pressing "Display".

I don't really understand why there is a difference between Data Layer and creating Polygons via google.maps.Polygon or the DrawingManager. Again, I'm fairly new to Google Maps.  My main reason for using the Data Layer because of the use and simplicity of GeoJSON.

**Comment by xkjyeah** [comment]
Agreed that GeoJSON is the universal language for most data applications.

This package was developed to wrap around the Google Maps API.

I actually doubt there is significant difference between Data Layer and using a computed property, because the Google Maps API is non-GL. At least performance-wise, I don't expect them to perform differently.
